Library/Homebrew/cmd/update-report.rb

```diff
@@ -136,8 +136,28 @@ def update_report
           DescriptionCacheStore.new(db)
                                .update_from_report!(hub)
         end
+
+        unless args.preinstall?
+          outdated_formulae = Formula.installed.count(&:outdated?)
+          outdated_casks = Cask::Caskroom.casks.count(&:outdated?)
+          msg = ""
+          if outdated_formulae.positive?
+            msg += "#{Tty.bold}#{outdated_formulae}#{Tty.reset} outdated #{"formula".pluralize(outdated_formulae)}"
+          end
+          if outdated_casks.positive?
+            msg += " and " if msg.present?
+            msg += "#{Tty.bold}#{outdated_casks}#{Tty.reset} outdated #{"cask".pluralize(outdated_casks)}"
+          end
+          if msg.present?
+            puts_stdout_or_stderr
+            puts_stdout_or_stderr <<~EOS
+              You have #{msg} installed.
+              You can update them with #{Tty.bold}brew upgrade#{Tty.reset}.
+            EOS
+          end
+        end
       end
-      puts if args.preinstall?
+      puts_stdout_or_stderr if args.preinstall?
```

Comment:
Could combine this with the `unless`

Reply:
This is positioned so that it also runs in the `hub.empty?` case - the `unless` block is not. Maybe doesn't matter much.

Reply:
Sorry, missed that.